### ISSUE TITLE
performance drops when i make some changes in SSH

### ISSUE THREAD

**Issue by takecareofbigboss** [mahyarnajibi/SSH#10]
hi, sth bothers me is that when i make some changes in ur code (these changes work in other face detection network), the performance drops.
any tricks in ur code?
and i wonder why iter=21000, the model achieves the best performance. above 21000, it drops.

hope i can get ur reply asap.

**Comment by takecareofbigboss** [comment]
@mahyarnajibi

**Comment by mahyarnajibi** [comment]
Thats because the validation set is used to determine the model hyper-parameters. Specifically, the training is stopped when a drop occurred on the validation set performance. If you continue training any detector after a while the performance starts to drop on the validation set because of overfitting on training. 
The final performance of the detectors should be compared together on the independent test set. For test set, we submitted the detections of the released model only once to be evaluated.
Hope I could understand your question correctly.

**Comment by takecareofbigboss** [comment]
maybe you can change ur batchsize to 12 by modifying ur code (resize and crop 500x500 of 12 images), the performance will boost. @mahyarnajibi

**Comment by po0ya** [comment]
@takecareofbigboss If you have tried this trick, please fork the project, make this change, report the numbers, and create a pull request. However, it will not be this easy, since we are detecting faces for anchors of size 512x512 from Module M3. So you should probably use 600x600 crops and also make sure you crop around the big faces to train M3 well.

**Comment by takecareofbigboss** [comment]
@mahyarnajibi  then, you can change 'scales' and 'base_size' to adapt this.

**Comment by takecareofbigboss** [comment]
after i finish this project, i will pull request.
